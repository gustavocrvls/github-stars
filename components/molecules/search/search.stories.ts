import type { Meta, StoryObj } from '@storybook/react';

import { Search } from './search';

const meta: Meta<typeof Search> = {
  title: 'Component/Molecules/Search',
  component: Search,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof Search>;

export const Default: Story = {};
