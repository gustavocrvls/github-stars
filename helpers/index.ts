export * from './render-with-provider';
