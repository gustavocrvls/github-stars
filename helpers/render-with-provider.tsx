import { render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles';

interface ProvidersWrapperProps {
  children: JSX.Element;
}

const ProvidersWrapper: React.FC<ProvidersWrapperProps> = ({ children }) => (
  <>
    {/** @ts-ignore */}
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </>
);

export const renderWithProvider = (component: JSX.Element, options?: any) => {
  return render(component, { wrapper: ProvidersWrapper, ...options });
};
