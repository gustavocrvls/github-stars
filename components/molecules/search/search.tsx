import { FormEvent } from 'react';
import { FiSearch } from 'react-icons/fi';
import { Container } from './search.styles';
import { Input } from 'components/atoms/input';
import { Button } from 'components/atoms/button';

interface Props {
  field: string;
  handleField: (search: string) => void;
  handleSearch: () => void;
}

export const Search = ({
  field,
  handleField,
  handleSearch,
}: Props): JSX.Element => {
  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    handleSearch();
  };

  return (
    <Container onSubmit={handleSubmit}>
      <Input
        placeholder="ex: react"
        value={field}
        onChange={e => handleField(e.target.value)}
      />
      <Button type="submit">
        <FiSearch size="18" />
      </Button>
    </Container>
  );
};
