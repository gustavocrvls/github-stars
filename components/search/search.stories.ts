import type { Meta, StoryObj } from '@storybook/react';

import { Search } from './search';

const meta: Meta<typeof Search> = {
  title: 'Component/Search',
  component: Search,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof Search>;

export const Default: Story = {};
