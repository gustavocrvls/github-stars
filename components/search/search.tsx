import { FormEvent } from 'react';
import { FiSearch } from 'react-icons/fi';
import { Container } from './search.styles';

interface Props {
  field: string;
  handleField: (search: string) => void;
  handleSearch: () => void;
}

export const Search = ({
  field,
  handleField,
  handleSearch,
}: Props): JSX.Element => {
  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    handleSearch();
  };

  return (
    <Container onSubmit={handleSubmit}>
      <input
        placeholder="ex: react"
        value={field}
        onChange={e => handleField(e.target.value)}
      />
      <button type="submit">
        <FiSearch size="18" />
      </button>
    </Container>
  );
};
