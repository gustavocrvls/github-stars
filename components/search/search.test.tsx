import { act } from 'react-dom/test-utils';
import { Search } from './search';
import { renderWithProvider } from 'helpers';

const handleField = jest.fn();
const handleSearch = jest.fn();

describe('Search', () => {
  it('Should match snapshot', () => {
    const view = renderWithProvider(
      <Search
        field="test"
        handleField={handleField}
        handleSearch={handleSearch}
      />,
    );
    expect(view).toMatchSnapshot();
  });

  it('Should use handleSearch on submit', async () => {
    const view = renderWithProvider(
      <Search
        field="test"
        handleField={handleField}
        handleSearch={handleSearch}
      />,
    );

    await act(async () => {
      const btn = await view.findByRole('button');
      btn.click();
    });

    expect(handleSearch).toHaveBeenCalledTimes(1);
  });
});
