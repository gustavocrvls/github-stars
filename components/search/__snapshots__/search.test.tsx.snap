// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-beqWaB kXpnnP"
      >
        <input
          placeholder="ex: react"
          value="test"
        />
        <button
          type="submit"
        >
          <svg
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-beqWaB kXpnnP"
    >
      <input
        placeholder="ex: react"
        value="test"
      />
      <button
        type="submit"
      >
        <svg
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
